### PR TITLE
rate limiter: interface to update limits & tests

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal @rf

### CODE REVIEW

File: node/rate_limiter.js [hunk 1]
```diff
@@ -78,7 +78,10 @@ function RateLimiter(options) {
     assert(self.numOfBuckets > 0 && self.numOfBuckets <= 1000, 'counter numOfBuckets should between (0 1000]');
 
     self.defaultServiceRpsLimit = options.defaultServiceRpsLimit || DEFAULT_SERVICE_RPS_LIMIT;
-    self.totalRpsLimit = options.totalRpsLimit || DEFAULT_TOTAL_RPS_LIMIT;
+    self.totalRpsLimit = options.totalRpsLimit;
+    if (!self.totalRpsLimit && self.totalRpsLimit !== 0) {
```
Comment (Raynos) [inline node/rate_limiter.js]: `if (typeof self.totalRpsLimit !== 'number')`

**Comment by Raynos** [comment]
lgtm.

File: node/test/rate-limiter.js [hunk 2]
```diff
@@ -163,3 +161,32 @@ test('rate exempt service works', function (assert) {
...
+    increment(rateLimiter, 'steve', 'bob');
+
+    assert.ok(rateLimiter.shouldRateLimitTotalRequest(), 'should rate limit total');
+    assert.ok(rateLimiter.shouldRateLimitService('steve'), 'should rate limit steve');
+    assert.ok(rateLimiter.shouldRateLimitService('bob'), 'should rate limit bob');
+
+    rateLimiter.updateTotalLimit(10);
+    rateLimiter.updateServiceLimit('steve', 10);
+
+    assert.ok(!rateLimiter.shouldRateLimitTotalRequest(), 'should not rate limit total');
```
Comment (kriskowal) [inline node/test/rate-limiter.js]: This should be shouldReateLimitTotalRequest**_s_** plural

**Comment by kriskowal** [comment]
lgtm as well

**Pull request merged**